src/components/StatusBadge/StatusBadge.tsx

```diff
@@ -34,17 +42,20 @@ export interface StatusBadgeProps {
  */
 export const StatusBadge: ChakraComponent<
   React.ForwardRefExoticComponent<
-    StatusBadgeProps & {
-      children?: React.ReactNode;
-    } & React.RefAttributes<HTMLDivElement>
+    React.PropsWithChildren<StatusBadgeProps> &
+      React.RefAttributes<HTMLDivElement>
   >,
-  StatusBadgeProps
+  React.PropsWithChildren<StatusBadgeProps>
 > = chakra(
   forwardRef<HTMLDivElement, React.PropsWithChildren<StatusBadgeProps>>(
     (props, ref?) => {
       const { children, className, id, level, type, ...rest } = props;
+      const labelFontSize = rest["fontSize"] || "desktop.body.body2";
```

Comment:
`"desktop.body.body2"` is also declared in the theme file. I think it makes more sense to not have this default here and instead just pass the value directly:
```
  labelFontSize: rest["fontSize"],
  variant: finalVariant,
```

This way, the default value is only declared once in one file and in the theme file which is more relevant than the react component.